server/src/main/java/org/opensearch/common/settings/ClusterSettings.java

```diff
@@ -646,7 +646,7 @@ public void apply(Settings value, Settings current, Settings previous) {
     public static final Map<String, List<Setting>> FEATURE_FLAGGED_CLUSTER_SETTINGS = Map.of(
         FeatureFlags.SEARCHABLE_SNAPSHOT,
         List.of(Node.NODE_SEARCH_CACHE_SIZE_SETTING),
-        FeatureFlags.REPLICATION_TYPE,
+        FeatureFlags.SEGMENT_REPLICATION_CLUSTER_SETTING,
```

Comment:
I thought the goal was to reuse the `FeatureFlags.REPLICATION_TYPE_SETTING` since we were going to get rid of the gating mechanism in other places, and keep this one behind the existing flag.

Reply:
Yes initially even I was thinking on same line,  but the reason I added new feature flag is because `FeatureFlags.REPLICATION_TYPE_SETTING` feature flag is old, I mean many users are already aware of this feature flag, they can mistakenly enable this expecting some other behaviour and some users might get confused on why after segrep GA release they have a feature flag. In order to avoid this confusion I thought creating a new feature flag which is not popular among users and only known to few for testing purposes.

Other than avoiding confusion I don't have any strong opinion on adding new feature flag. @kotwanikunal what are your thoughts?

Reply:
Considering new features and enhancements that we are going to introduce in the future, I think it better to use the same feature flag, So, they are flagged as beta features and users can use same flag to test them, instead of having different flags for each enhancement.

Reply:
That was my original intent. An alternative solution that might work going forward is introducing a new feature flag which gates experimental features only - `FeatureFlags.SEGMENT_REPLICATION_EXPERIMENTAL`. This will avoid the confusion that @Rishikesh1159 was talking about as well as serves our purpose going forward with new/experimental features until we release them.

Reply:
Sure thanks @kotwanikunal and @anasalkouz for your thoughts. I think kunal's alternative solution here makes sense. It will make things cleaner. I will go ahead and create new FeatureFlags.SEGMENT_REPLICATION_EXPERIMENTAL and we can use this for future segrep experimental features.